diff --git a/where_is_my_sddm_theme/Main.qml b/where_is_my_sddm_theme/Main.qml
index a9089fa..cd97514 100644
--- a/where_is_my_sddm_theme/Main.qml
+++ b/where_is_my_sddm_theme/Main.qml
@@ -92,7 +92,7 @@ Rectangle {
                 id: image
                 anchors.fill: parent
                 source: config.background
-                fillMode: Image.PreserveAspectCrop
+                fillMode: Image.Pad
                 z: 2
               }
 
